code/src/java/plugin/function/InputFunction.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2016 (C) Tom Parker <thpr@users.sourceforge.net>
+ * 
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ * 
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+package plugin.function;
+
+import java.util.Arrays;
+
+import pcgen.base.formula.base.DependencyManager;
+import pcgen.base.formula.base.EvaluationManager;
+import pcgen.base.formula.base.FormulaSemantics;
+import pcgen.base.formula.base.Function;
+import pcgen.base.formula.base.LegalScope;
+import pcgen.base.formula.base.VariableLibrary;
+import pcgen.base.formula.parse.ASTQuotString;
+import pcgen.base.formula.parse.Node;
+import pcgen.base.formula.parse.SimpleNode;
+import pcgen.base.formula.visitor.DependencyVisitor;
+import pcgen.base.formula.visitor.EvaluateVisitor;
+import pcgen.base.formula.visitor.SemanticsVisitor;
+import pcgen.base.formula.visitor.StaticVisitor;
+import pcgen.base.util.FormatManager;
+import pcgen.output.channel.ChannelUtilities;
+
+public class InputFunction implements Function
+{
+
+	/**
+	 * Returns the function name for this function. This is how it is called by
+	 * a user in a formula.
+	 * 
+	 * @see pcgen.base.formula.base.Function#getFunctionName()
+	 */
+	@Override
+	public String getFunctionName()
+	{
+		return "INPUT";
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public final FormatManager<?> allowArgs(SemanticsVisitor visitor, Node[] args,
+		FormulaSemantics semantics)
+	{
+		int argCount = args.length;
+		if (argCount != 1)
+		{
+			semantics.setInvalid("Function " + getFunctionName()
+				+ " received incorrect # of arguments, expected: 1 got "
+				+ args.length + " " + Arrays.asList(args));
+			return null;
```

Comment:
So the entire concept of how allowArgs works depends on returning null... if you want to change that, it's a -formula library change we can discuss